zk_toolbox/crates/zk_inception/src/commands/consensus.rs

```diff
@@ -0,0 +1,321 @@
+    let b: [u8; 96] = ByteFmt::encode(k).try_into().unwrap();
+    abi::Bls12381PublicKey {
+        a: b[0..32].try_into().unwrap(),
+        b: b[32..64].try_into().unwrap(),
+        c: b[64..96].try_into().unwrap(),
+    }
+}
+
+fn encode_validator_pop(pop: &validator::ProofOfPossession) -> abi::Bls12381Signature {
+    let b: [u8; 48] = ByteFmt::encode(pop).try_into().unwrap();
+    abi::Bls12381Signature {
+        a: b[0..32].try_into().unwrap(),
+        b: b[32..48].try_into().unwrap(),
+    }
+}
+
+#[derive(clap::Subcommand, Debug)]
+pub enum Command {
+    /// Sets the attester committee in the consensus registry contract to
+    /// `consensus.genesis_spec.attesters` in general.yaml.
+    SetAttesterCommittee,
+    /// Fetches the attester committee from the consensus registry contract.
+    GetAttesterCommittee,
+}
+
+/// Collection of sent transactions.
+#[derive(Default)]
+pub struct TxSet(Vec<(H256, &'static str)>);
+
+impl TxSet {
+    /// Sends a transactions and stores the transaction hash.
+    pub async fn send<M: 'static + Middleware, B: Borrow<M>, D: Detokenize>(
+        &mut self,
+        name: &'static str,
+        call: FunctionCall<B, M, D>,
+    ) -> anyhow::Result<()> {
+        let h = call.send().await.context(name)?.tx_hash();
+        self.0.push((h, name));
+        Ok(())
+    }
+
+    /// Waits for all stored transactions to complete.
+    pub async fn wait<P: JsonRpcClient>(self, provider: &Provider<P>) -> anyhow::Result<()> {
+        for (h, name) in self.0 {
+            async {
+                let status = PendingTransaction::new(h, provider)
+                    .await
+                    .context("await")?
+                    .context("receipt missing")?
+                    .status
+                    .context("status missing")?;
+                anyhow::ensure!(status == 1.into(), "transaction failed");
+                Ok(())
+            }
+            .await
+            .context(name)?;
+        }
+        Ok(())
+    }
+}
+
+async fn get_attester_committee(
+    consensus_registry: &abi::ConsensusRegistry<impl 'static + Middleware>,
+) -> anyhow::Result<attester::Committee> {
+    let attesters = consensus_registry
+        .get_attester_committee()
+        .call()
+        .await
+        .context("get_attester_committee()")?;
+    let attesters: Vec<_> = attesters
+        .iter()
+        .map(decode_weighted_attester)
+        .collect::<Result<_, _>>()
+        .context("decode_weighted_attester")?;
+    let attesters =
+        attester::Committee::new(attesters.into_iter()).context("attester::Committee::new()")?;
+    Ok(attesters)
+}
+
+fn print_attesters(committee: &attester::Committee) {
+    logger::success(
+        committee
+            .iter()
+            .map(|a| format!("{a:?}"))
+            .collect::<Vec<_>>()
+            .join("\n"),
+    );
+}
+
+impl Command {
+    pub(crate) async fn run(self, shell: &Shell) -> anyhow::Result<()> {
```

Comment:
nit: this function looks extremely complex, could you split it into more small ones?

Reply:
done